packages/nextjs/components/scaffold-eth/EtherInput.tsx

```diff
@@ -14,69 +46,45 @@ type TEtherInputProps = {
  *
  * onChange will always be called with the value in ETH
  */
-export default function EtherInput({ value, name, placeholder, onChange }: TEtherInputProps) {
-  const [ethValue, setEthValue] = useState("");
-  const [displayValue, setDisplayValue] = useState("");
-  const [usdMode, setUSDMode] = useState(false);
-
+export default function EtherInput({ value = "", name, placeholder, onChange }: EtherInputProps) {
+  const [transitoryDisplayValue, setTransitoryDisplayValue] = useState<string>();
   const ethPrice = useAppStore(state => state.ethPrice);
+  const [usdMode, setUSDMode] = useState(false);
 
-  // Controlled vs Uncontrolled input.
-  const currentValue = value !== undefined ? value : ethValue;
-
-  useEffect(() => {
-    // Reset when clearing controlled input
-    if (!currentValue) {
-      setDisplayValue("");
-      setEthValue("");
-    }
-  }, [currentValue]);
+  // The displayValue is derived from the ether value that is controlled outside of the component
+  // In usdMode, it is converted to its usd value, in regular mode it is unaltered
+  const displayValue = useMemo(() => {
+    // Clear any transitory display values that might be set
+    setTransitoryDisplayValue(undefined);
+    return etherValueToDisplayValue(usdMode, value, ethPrice);
+  }, [ethPrice, usdMode, value]);
 
-  const onChangeNumber = (event: ChangeEvent<HTMLInputElement>) => {
-    let ethNewValue;
+  const handleChangeNumber = (event: ChangeEvent<HTMLInputElement>) => {
     const newValue = event.target.value;
 
+    // Following condition is a fix to prevent usdMode from experiencing different display values
+    // than what the user entered. This can happen due to floating point rounding errors that are introduced in the back and forth conversion
     if (usdMode) {
-      const parsedNewValue = parseFloat(newValue);
-      if (Number.isNaN(parsedNewValue)) {
-        // Invalid number.
-        ethNewValue = newValue;
-      } else {
-        // Compute the ETH value if a valid number.
-        ethNewValue = (parsedNewValue / ethPrice).toString();
+      const decimals = newValue.split(".")[1];
+      if (decimals && decimals.length > MAX_DECIMALS_USD) {
+        return;
       }
-    } else {
-      ethNewValue = newValue;
     }
 
-    setEthValue(ethNewValue);
-    setDisplayValue(newValue);
-    if (onChange) {
-      onChange(ethNewValue);
+    // Since the display value is a derived state (calculated from the ether value), usdMode would not allow introducing a decimal point.
+    // This condition handles a transitory state for a display value with a trailing decimal sign
+    if (newValue.endsWith(".")) {
```

Comment:
` || newValue.endsWith(".0")` or you can't write for example $ 1.02

Reply:
Ah good catch :+1: